### ISSUE TITLE
Chowning the storage directory uses the wrong user and possibly the wrong directory

### ISSUE THREAD

**Issue by jarshwah** [echocat/puppet-graphite#156]
I manage my own web server, and I'm using nginx/uwsgi to do it. The nginx user runs everything. This setup worked in version 5.5 of the module.

The chown is forcing the apache user to be the owner of the storage directory. The following block from `params.pp` isn't working as I'd expect:

```
if $graphite::gr_web_user {
    $web_user = $graphite::gr_web_user
} else {
    $web_user = 'apache'
}
```

https://github.com/echocat/puppet-graphite/blob/0c02ae9a5be24b96cdcff782dc6dd5f9ab9eb221/manifests/config.pp#L62

The above also hardcodes the storage directory. The whisper directory may be relocated, and I'm not sure if the web user needs to have owner/group permissions on the relocated whisper directory or not.

my yaml:

```yaml
graphite::gr_web_server: 'none' # we will do it
graphite::gr_user: 'nginx'
graphite::gr_group: 'nginx'
graphite::gr_web_user: 'nginx'
graphite::gr_web_group: 'nginx'
graphite::params::web_user: 'nginx'  # doesnt seem to work
graphite::params::web_group: 'nginx'  # doesnt seem to work
graphite::gr_local_data_dir: '/data/whisper'
```

**Comment by dwerder** [comment]
Ok, the relocated storage directory and the hardcoded chown could be a bug.

But as your yaml file describes it correctly, you should use `graphite::gr_web_group` and `graphite::gr_web_user` instead of changing things in params.pp

**Comment by jarshwah** [comment]
Sorry, I am trying to use gr_web_user and gr_web_group. I believe I had to try to modify params::web_user for 5.5. Removing those two entries in hiera doesn't fix the problem.

2 of my hosts are using the correct "nginx" user to chown. 1 of my hosts is not (chowning as apache user), though, which is really strange. So the bug is intermittent.

Trying to debug hiera from the master shows that params.pp is trying to access variables from init.pp, which can't work because params has to be evaluated before init:

```
puppet master --debug --compile bad.host.com | grep hiera
```

> Warning: Scope(Class[Graphite::Params]): Could not look up qualified variable 'graphite::gr_web_group'; class graphite has not been evaluated
Warning: Scope(Class[Graphite::Params]): Could not look up qualified variable 'graphite::gr_web_user'; class graphite has not been evaluated

Changing:

```
if $graphite::gr_web_user {
    $web_user = $graphite::gr_web_user
} else {
    $web_user = 'apache'
}
```

To:

if $::graphite::gr_web_user {
    $web_user = $::graphite::gr_web_user
} else {
    $web_user = 'apache'
}

Shows this instead:

> Warning: Scope(Class[Graphite::Params]): Could not look up qualified variable '::graphite::gr_web_group'; class ::graphite has not been evaluated
Warning: Scope(Class[Graphite::Params]): Could not look up qualified variable '::graphite::gr_web_user'; class ::graphite has not been evaluated

Which shows this is the problem line.

`graphite` inherits from `params`. `params` must run first, but `params` attempts to access variables within `class graphite`. This will always end up using the default value, except in certain configurations that I'm unable to figure out.

How I've worked around this in another project is by assigning the default parameters directly in the constructor of the main class:

https://github.com/Engage/puppet-uwsgi/blob/master/manifests/init.pp#L90

```
class graphite {
   ...
   gr_web_user = $::graphite::params::web_user
   ...
}
```

But I'm not entirely sure that there aren't problems with this approach too.

Sorry if I've rambled a bit, but I was debugging and understanding more as I was writing this. Let me know if I've gone too far off course and need to explain more clearly.
